src/huggingface_hub/hf_api.py

```diff
@@ -8261,6 +8437,137 @@ def grant_access(
+
+    @validate_hf_hub_args
+    def list_webhooks(self, token: Optional[str] = None) -> List[Dict]:
+        """List all configured webhooks."""
+        response = get_session().get(
+            f"{ENDPOINT}/api/settings/webhooks",
+            headers=self._build_hf_headers(token=token),
+        )
+        hf_raise_for_status(response)
+        return response.json()
+
+    @validate_hf_hub_args
+    def create_webhook(
+        self,
+        watched: List[WatchedItem],
+        url: str,
+        domains: List[WEBHOOK_DOMAIN_T],
+        secret: Optional[str] = None,
+        token: Optional[str] = None,
+    ) -> Dict:
+        """Create a new webhook.
+
+        Args:
+            watched (List[WatchedItem]):
+                List of items to watch. It an be users, orgs, models, datasets or spaces.
+                See `WatchedItem` for more details.
+            url (str):
+                URL to send the payload to.
+            domains (List[Literal["repo", "discussion"]]):
+                List of domains to watch. It can be "repo", "discussion" or both.
+            secret (`str`, *optional*):
+                A secret to sign the payload with.
+            token (`str`, *optional*):
+                A valid authentication token (see https://huggingface.co/settings/token).
+
+        Returns:
+            dict: The created webhook.
+
+        Example:
+            ```python
+            >>> payload = create_webhook(
+            ...     watched=[{"type": "user", "name": "julien-c"}, {"type": "org", "name": "HuggingFaceH4"}],
+            ...     url="https://webhook.site/a2176e82-5720-43ee-9e06-f91cb4c91548",
+            ...     domains=["repo", "discussion"],
+            ...     secret="my-secret",
+            ... )
+            {
+                "webhook": {
+                    "id": "654bbbc16f2ec14d77f109cc",
+                    "watched": [{"type": "user", "name": "julien-c"}, {"type": "org", "name": "HuggingFaceH4"}],
+                    "url": "https://webhook.site/a2176e82-5720-43ee-9e06-f91cb4c91548",
+                    "token": "my-token",
+                    "domains": ["repo", "discussion"],
+                    "disabled": False,
+                },
+            }
+            ```
+        """
+        response = get_session().post(
+            f"{ENDPOINT}/api/settings/webhooks",
+            json={"watched": watched, "url": url, "domains": domains, "secret": secret},
+            headers=self._build_hf_headers(token=token),
+        )
+        hf_raise_for_status(response)
+        return response.json()
+
+    @validate_hf_hub_args
+    def update_webhook(
+        self,
+        webhook_id: str,
+        watched: List[WatchedItem],
+        url: str,
+        domains: List[WEBHOOK_DOMAIN_T],
+        secret: Optional[str] = None,
+        token: Optional[str] = None,
+    ) -> Dict:
+        """Update an existing webhook.
+
+        Exact same usage as `create_webhook` but you must know the `webhook_id`.
```

Comment:
It's best to duplicate the `args` section here. This is extra maintenance work but very helpful for IDEs and to correctly generate the documentation.

Reply:
I've updated the docstring as suggested in the upcoming commit.

I'll proceed with the remaining updates once you've reviewed the implementation.